src/cpp/statistics/rtps/StatisticsBase.hpp

```diff
@@ -282,21 +293,46 @@ class StatisticsParticipantImpl
             const uint32_t packages);
 
     /*
-     * Report PDP message exchange.
-     * We filtered the non-pdp traffic here to minimize presence of statistics code in endpoints implementation.
+     * Auxiliary method to report EDP message exchange.
+     * @param packages number of edp packages sent
+     */
+    void on_edp_packet(
+            const uint32_t packages);
+
+    /*
+     * Report discovery protocols message exchange.
+     * We filtered the non discovery traffic here to minimize presence of statistics code in endpoints implementation.
      * @param sender_guid GUID_t to filter
-     * @param destination_locators_begin, start of locators range
-     * @param destination_locators_end, end of locators range
+     * @param destination_locators_begin start of locators range
+     * @param destination_locators_end end of locators range
      */
     template<class LocatorIteratorT>
-    void on_pdp_packet(
+    void on_dp_packet(
             const GUID_t& sender_guid,
             const LocatorIteratorT& destination_locators_begin,
             const LocatorIteratorT& destination_locators_end)
     {
-        if (sender_guid.entityId == fastrtps::rtps::c_EntityId_SPDPWriter
-                && destination_locators_begin != destination_locators_end)
+        if ( destination_locators_begin != destination_locators_end )
         {
+            void (StatisticsParticipantImpl::* discovery_callback)(
+                    const uint32_t) = nullptr;
+
+            switch (sender_guid.entityId.to_uint32())
+            {
+                case ENTITYID_SPDP_BUILTIN_RTPSParticipant_WRITER:
+                case ENTITYID_SPDP_BUILTIN_RTPSParticipant_READER:
+                    discovery_callback = &StatisticsParticipantImpl::on_pdp_packet;
+                    break;
+                case ENTITYID_SEDP_BUILTIN_PUBLICATIONS_WRITER:
+                case ENTITYID_SEDP_BUILTIN_PUBLICATIONS_READER:
+                case ENTITYID_SEDP_BUILTIN_SUBSCRIPTIONS_WRITER:
+                case ENTITYID_SEDP_BUILTIN_SUBSCRIPTIONS_READER:
```

Comment:
When building with security, [these ids](https://github.com/eProsima/Fast-DDS/blob/af466cfe63a8319cc9d37514267de8952627a9a4/include/fastdds/rtps/common/EntityId_t.hpp#L55-L58) should also be accounted.

Reply:
didn't notice that